sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_receiver_async.py

```diff
@@ -54,13 +82,21 @@ async def _incoming_attach(self, frame):
         self.current_link_credit = self.link_credit
         await self._outgoing_flow()
 
+    async def _incoming_flow(self, frame):
+        drain = frame[8]  # drain
+        # If we have sent an outgoing flow frame with drain, wait for the response
+        if self._drain_state and drain:
+            self._drain_state = False
+            self._received_drain_response = True
```

Comment:
This is another place where we'll need to confirm this is concurrency-safe

Reply:
Not that the links are concurrency-safe at the moment - but it's a problem we need to solve before pyamqp can be published, so we should keep that in mind when new code is added.